### PR TITLE
Add complete the EmployeeDocumentControllerUnitTest

### PR DESCRIPTION
**Pull request by KAYEMBAJONATHAN**
- Add complete the Employee Document Controller 
Screenshot 2024-06-07 081052
Unit Test

### CODE REVIEW

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 1]
```diff
@@ -269,4 +270,33 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
         Assert.Equal("An error occurred while fetching the employee documents.", notfoundResult.Value);
 
     }
+
+    [Fact]
+    public async Task GetAllDocuments_ReturnsInternalServerError_WhenExceptionIsThrown()
+    {
+        _employeeMockDocumentService
+                .Setup(ex => ex.GetAllDocuments())
```
Comment (bseptember) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:278]:
```suggestion
                .Setup(service => service.GetAllDocuments())
```

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 2]
```diff
@@ -269,4 +270,33 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
...
+
+        var statusCodeResult = Assert.IsType<ObjectResult>(result);
+        Assert.Equal(500, statusCodeResult.StatusCode);
+        Assert.Equal("An error occurred while fetching the employee documents.", statusCodeResult.Value);
+    }
+
+    [Fact]
+    public async Task GetAllDocuments_ReturnsOk_WhenDocumentsAreFetchedSuccessfully()
+    {
+        var documents = new List<SimpleEmployeeDocumentGetAllDto> { };
```
Comment (bseptember) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:291]:
```suggestion
        var documents = new List<SimpleEmployeeDocumentGetAllDto>();
```

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 3]
```diff
@@ -269,4 +270,33 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
...
+        _employeeMockDocumentService
+            .Setup(ex => ex.GetAllDocuments())
```
Comment (bseptember) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:293]:
```suggestion
            .Setup(service => service.GetAllDocuments())
```

**Review by bseptember — DISMISSED** [review]

**Comment by KAYEMBAJONATHAN** [comment]
Screenshot 2024-06-07 084019
fix

**Comment by dvanhout89** [comment]
Please format your entire unit tests. 

- Remove unused imports.
- Fix Naming of Methods.
- Fix Line spacing and White spacing.
- Remove ghost code //commented out code.

**Review by dvanhout89 — CHANGES_REQUESTED** [review]
Please format your entire unit tests.

- Remove unused imports.
- Fix Naming of Methods.
- Fix Line spacing and White spacing.
- Remove ghost code //commented out code.

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 4]
```diff
@@ -269,4 +270,33 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
         Assert.Equal("An error occurred while fetching the employee documents.", notfoundResult.Value);
 
     }
+
+    [Fact]
+    public async Task GetAllDocuments_ReturnsInternalServerError_WhenExceptionIsThrown()
+    {
+        _employeeMockDocumentService
+                .Setup(x => x.GetAllDocuments())
```
Comment (bseptember) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:278]:
```suggestion
                .Setup(service => service.GetAllDocuments())
```

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 5]
```diff
@@ -269,4 +270,33 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
...
+        Assert.Equal(500, statusCodeResult.StatusCode);
+        Assert.Equal("An error occurred while fetching the employee documents.", statusCodeResult.Value);
+    }
+
+    [Fact]
+    public async Task GetAllDocuments_ReturnsOk_WhenDocumentsAreFetchedSuccessfully()
+    {
+        var documents = new List<SimpleEmployeeDocumentGetAllDto>();
+        _employeeMockDocumentService
+            .Setup(x => x.GetAllDocuments())
```
Comment (bseptember) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:293]:
```suggestion
            .Setup(service => service.GetAllDocuments())
```

**Review by bseptember — APPROVED** [review]

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 6]
```diff
@@ -267,6 +258,34 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
...
+
+        var statusCodeResult = Assert.IsType<ObjectResult>(result);
+        Assert.Equal(500, statusCodeResult.StatusCode);
+        Assert.Equal("An error occurred while fetching the employee documents.", statusCodeResult.Value);
+    }
+
+    [Fact]
+    public async Task GetAllDocuments_ReturnsOk_WhenDocumentsAreFetchedSuccessfully()
+    {
+        var documents = new List<SimpleEmployeeDocumentGetAllDto>();
```
Comment (LlewellynAnthony) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:280]: Create a list that contains documents, and assert that all the documents in the populated list are being fetched.

**Review by dvanhout89 — CHANGES_REQUESTED** [review]
Please format your entire unit tests.

- [x] Remove unused imports.
- [ ] Fix Naming of Methods.
- [ ] Fix Line spacing and White spacing.
- [ ] Remove ghost code //commented out code.

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs
Comment (MatthewSchoeman) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:1]: On line 96 in this file, there is a list of EmployeeDocuments that is created but it is never used in the method. If it has no purpose, then it should be removed

**Review by dvanhout89 — CHANGES_REQUESTED** [review]
**Line 278**
```csharp
GetAllDocuments_ReturnsOk_When**No**DocumentsAreFetched()
```
Because:
**Line 280**
// Arrange: Setup the mock service to return an empty list of documents
```csharp
var documents = new List<SimpleEmployeeDocumentGetAllDto>();
```

Then:
```csharp
[Fact]
public async Task GetAllDocumentsReturnsOkWhenDocumentsAreFetched()
{
    // Arrange
    var documents = new List<SimpleEmployeeDocumentGetAllDto>
    {
        new SimpleEmployeeDocumentGetAllDto { Id = 1, FileName = "Doc1", /* other properties */ },
        new SimpleEmployeeDocumentGetAllDto { Id = 2, FileName = "Doc2", /* other properties */ }
    };
    ``

**Comment by KAYEMBAJONATHAN** [comment]
Screenshot 2024-06-07 103929

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 7]
```diff
@@ -172,23 +161,8 @@ public async Task UpdateEmployeeDocumentReturnsOkResult()
     [Fact]
     public async Task UpdateEmployeeDocumentReturnsNotFoundResultWhenExceptionThrown()
     {
-
         EmployeeDocumentDto broken = new EmployeeDocumentDto
         {
-            Id = 1,
```
Comment (LlewellynAnthony) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:178]: Why is this code removed? One must be able to assert if the values are being updated. And if this variable is not being used in this test, remove it entirely

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 8]
```diff
@@ -212,7 +186,6 @@ public async Task DeleteEmployeeDocumentsReturnsOkResult()
         var actualemployeeDocument = Assert.IsAssignableFrom<EmployeeDocumentDto>(okResult.Value);
```
Comment (LlewellynAnthony) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:186]: Rename to actualEmployeeDocument

**Review by dvanhout89 — DISMISSED** [review]

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 9]
```diff
@@ -267,6 +253,37 @@ public async Task GetEmployeeDocumentByStatusReturnsNotFoundResultWhenExceptionT
...
+
+        var result = await _controller.GetAllDocuments();
+
+        var statusCodeResult = Assert.IsType<ObjectResult>(result);
+        Assert.Equal(500, statusCodeResult.StatusCode);
+        Assert.Equal("An error occurred while fetching the employee documents.", statusCodeResult.Value);
+    }
+
+    [Fact]
+    public async Task GetAllDocumentsReturnsOkWhenDocumentsAreFetched()
```
Comment (bseptember) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:273]:
```suggestion
    public async Task GetAllDocuments_ReturnsOk_WhenDocumentsAreFetched()
```

**Review by LlewellynAnthony — DISMISSED** [review]

**Review by bseptember — DISMISSED** [review]

**Review by bseptember — APPROVED** [review]

**Review by dvanhout89 — APPROVED** [review]

**Review by LlewellynAnthony — APPROVED** [review]

**Pull request merged**